### PR TITLE
fix: test failing because of etched deployment address

### PR DESCRIPTION
**Pull request by xBA5ED**
Every once in a while the foundry fuzzer will pass an address that is the next deployment address for the (pranked) `msg.sender`.

When this happens we etch the address so it passes the `noCode` check, however in some tests we deploy a contract that will attempt to deploy to this same address, this causes a revert.

This fix resolves this and bumps forge-std so we can use the new `readCallers()` cheatcode.

### CODE REVIEW

File: test/unit/PostSettlementSurplusAuctionHouse.t.sol [hunk 1]
```diff
@@ -114,9 +114,7 @@ contract Unit_PostSettlementSurplusAuctionHouse_Constructor is Base {
     assertEq(address(postSettlementSurplusAuctionHouse.safeEngine()), _safeEngine);
   }
 
-  function test_Set_ProtocolToken(address _protocolToken) public happyPath {
-    vm.assume(uint160(_protocolToken) > 20);
-    vm.etch(_protocolToken, '0xF');
+  function test_Set_ProtocolToken(address _protocolToken) public happyPath mockAsContract(_protocolToken) {
```
Comment (0xJabberwock) [inline test/unit/PostSettlementSurplusAuctionHouse.t.sol:117]: Executing `mockAsContract` within the `happyPath` would make sense.

File: test/utils/HaiTest.t.sol [hunk 2]
```diff
@@ -133,6 +133,14 @@ contract OverflowChecker {
...
+    // Foundry fuzzer sometimes gives us the next deployment address
+    // this results in very unexpected reverts as any contract deploy will revert
+    // we check here to make sure its not the next deployment address for the (pranked) msg.sender
+    (, address _msgSender,) = vm.readCallers();
+    address _nextDeploymentAddr = computeCreateAddress(address(_msgSender), vm.getNonce(_msgSender));
+
+    vm.assume(_address != _nextDeploymentAddr);
+
     // It should not be a precompile
     vm.assume(uint160(_address) > 20);
```
Comment (0xJabberwock) [inline test/utils/HaiTest.t.sol:145]: `assumeNoPrecompiles()` forge-std cheat fits like a glove.

File: test/utils/HaiTest.t.sol [hunk 3]
```diff
@@ -133,6 +133,14 @@ contract OverflowChecker {
 
 abstract contract HaiTest is DSTestPlus, OverflowChecker {
   modifier mockAsContract(address _address) {
+    // Foundry fuzzer sometimes gives us the next deployment address
+    // this results in very unexpected reverts as any contract deploy will revert
+    // we check here to make sure its not the next deployment address for the (pranked) msg.sender
+    (, address _msgSender,) = vm.readCallers();
+    address _nextDeploymentAddr = computeCreateAddress(address(_msgSender), vm.getNonce(_msgSender));
+
+    vm.assume(_address != _nextDeploymentAddr);
```
Comment (0xJabberwock) [inline test/utils/HaiTest.t.sol:142]: Hmm. The mocked contract address may still collide with the address of a possible second deployment. Besides, the reversions may still happen upon deploying from currently inactive msg senders. 🤔

**Review by 0xJabberwock — COMMENTED** [review]
Nice catch! I left some thoughts.

File: test/utils/HaiTest.t.sol (hunk 2 above)
Comment (xBA5ED) [reply test/utils/HaiTest.t.sol:145]: I wasn't aware this existed, good call

**Review by 0xJabberwock — APPROVED** [review]

File: test/utils/HaiTest.t.sol (hunk 3 above)
Comment (wei3erHase) [reply test/utils/HaiTest.t.sol:142]: can u run a `yarn test:unit:deep` on that test and see if it collides? if it doesn't, i'm ok with it

Comment (0xJabberwock) [reply test/utils/HaiTest.t.sol:142]: I ran it a few times. It does not collide! 🎉

**Review by wei3erHase — APPROVED** [review]
👏

**Pull request merged**